src/Agent.Worker/ResourceMetricsManager.cs

```diff
@@ -286,65 +286,26 @@ private async Task GetMemoryInfoAsync()
 
             if (PlatformUtil.RunningOnLinux)
             {
-                // Some compact Linux distributions like UBI may not have "free" utility installed, or it may have a custom output
-                // We don't want to break currently existing pipelines with ADO warnings
-                // so related errors thrown here will be sent to the trace or debug logs by caller methods
-
-                using var processInvoker = HostContext.CreateService<IProcessInvoker>();
-
                 using var timeoutTokenSource = new CancellationTokenSource();
                 timeoutTokenSource.CancelAfter(TimeSpan.FromMilliseconds(METRICS_UPDATE_INTERVAL));
 
                 using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(
                     _context.CancellationToken,
                     timeoutTokenSource.Token);
 
-                processInvoker.OutputDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+                await Task.Run(() =>
                 {
-                    if (!message.Data.StartsWith("Mem:"))
-                    {
-                        return;
-                    }
-
-                    var processInvokerOutputString = message.Data;
-                    var memoryInfoString = processInvokerOutputString.Split(" ", StringSplitOptions.RemoveEmptyEntries);
-
-                    if (memoryInfoString.Length != 7)
-                    {
-                        throw new Exception("\"free\" utility has non-default output");
-                    }
+                    string memoryInfo = File.ReadAllText("/proc/meminfo");
```

Comment:
Is there any reason to use delegate task?
Consider just use the ReadAllTextAsync instead:
https://learn.microsoft.com/en-us/dotnet/api/system.io.file.readalltextasync?view=net-6.0

Reply:
This is the atomic operation which is wrapped in task due to cancelation logic. 
If we use only the ReadAllTextAsync with the token, in case of timeout it cancels the read operation but continue to execute the parse, so in this case we need to add additional exit code block. However, in case of using the task we hide all operations under one timeout condition, which is the most practical solution.

Reply:
Task cancellation works only when the work hasn't started yet
https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run?view=net-6.0#system-threading-tasks-task-run(system-func((system-threading-tasks-task))-system-threading-cancellationtoken)

Reply:
Yep, missed it. Fixed, thanks!